### ISSUE TITLE
Allow multiple units in one isSame() call

### ISSUE THREAD

**Issue by hanifbbz** [moment/moment#2234]
We use `isSame()` a lot in our unit tests. However we usually end up with code like this:

    test.ok(moment(results.date).isSame(testData[i].date, 'year'), 'year is correct');
    test.ok(moment(results.date).isSame(testData[i].date, 'month'), 'month is correct');
    test.ok(moment(results.date).isSame(testData[i].date, 'day'), 'day is correct');

This makes our test codes verbose. In a dream world we could do:

    test.ok(moment(results.date).isSame(testData[i].date, ['year', 'month', 'day']), 'date is correct');

**Comment by Jervelund** [comment]
For this specific case I think you can just use the most precise unit as the second parameter.

```
test.ok(moment(results.date).isSame(testData[i].date, 'day'), 'date is correct');
```

Verification:
```
a = moment();
b = moment().set('month',0); // a and b have same year and date, but different month.
a.isSame(b,'year'); // true (checks only year)
a.isSame(b,'month'); // false  (checks year and month)
a.isSame(b,'day'); // false (checks year, month and date)
```

**Comment by hanifbbz** [comment]
ok, makes sense. However I think the documentation is kinda vague about it and can be improved.

For example, the fact that by default it is comparing the milliseconds can be made more dominant in the sentence: "If you want to limit the granularity to a unit other than milliseconds, pass the units as the second parameter." (this only implies that the default is millisecond comparison)

Also it should be mentioned that `isSame('day')` actually does a check on `month` and `year` too.

In other words, `isSame('day')` is very different from doing `a.get('day') === b.get('day')`.

Maybe we can close this issue just by updating the documentation?

**Comment by Jervelund** [comment]
Agree - I'll add a note to the relevant functions and make a pull request.

**Comment by swbullis** [comment]
I had the same issue.  Another possibility to make this less confusing would be alias 'date' to 'day'
so it could read  something like
     moment("2015-02-08").isSame("2015-02-08 01:00 pm", 'date')

**Comment by hanifbbz** [comment]
@swbullis  :+1:

**Comment by mj1856** [comment]
Fixed by moment/momentjs.com/pull/176
